sci-libs/gsl/gsl-2.8.ebuild

```diff
@@ -0,0 +1,65 @@
+# Copyright 1999-2024 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+inherit autotools flag-o-matic toolchain-funcs
+
+DESCRIPTION="The GNU Scientific Library"
+HOMEPAGE="https://www.gnu.org/software/gsl/"
+SRC_URI="mirror://gnu/${PN}/${P}.tar.gz
+	https://github.com/Jamim/${PN}/commit/${P//./%2E}-cblas.patch -> ${P}-cblas.patch"
+
+LICENSE="GPL-3+"
+SLOT="0/$(ver_cut 1-2)"
+KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~hppa ~ia64 ~loong ~mips ~ppc ~ppc64 ~riscv ~s390 ~sparc ~x86 ~amd64-linux ~x86-linux"
+IUSE="cblas-external +deprecated static-libs"
+
+RDEPEND="cblas-external? ( virtual/cblas:= )"
+DEPEND="${RDEPEND}"
+BDEPEND="virtual/pkgconfig"
+
+PATCHES=(
+	"${DISTDIR}"/${P}-cblas.patch
+	"${FILESDIR}"/${PN}-2.7.1-test-tolerance.patch
+	"${FILESDIR}"/${PN}-drop-broken-rng-tests.patch
+)
+
+src_prepare() {
+	default
+
+	if use deprecated; then
+		sed -i -e "/GSL_DISABLE_DEPRECATED/,+2d" configure.ac || die
+	fi
+
+	eautoreconf
+}
+
+src_configure() {
+	filter-flags -ffast-math
+
+	# with -ffp-contract=fast tests are broken beyond repair
+	# which indicates that GSL might produce inaccurate values
+	append-flags -ffp-contract=off
```

Comment:
Interesting catch. Looking at the upstream bugtracker, I see https://savannah.gnu.org/bugs/?64613 so probably worth including the link as a comment in the ebuild.

(If I hadn't found an existing one, I would recommend you submit a bug report yourself.)